File: Win/ui/views/manager_dashboard_view.py
"""
ManagerDashboardView.py — DAS CRM Windows
Manager-Specific KPIs, Team Performance, and Subordinate Metrics
Feature parity with Android ManagerDashboardScreen.tsx
"""

from PyQt6.QtWidgets import (
    QWidget, QVBoxLayout, QHBoxLayout, QLabel, QPushButton, QScrollArea,
    QFrame, QGridLayout, QTableWidget, QTableWidgetItem, QAbstractItemView
)
from PyQt6.QtCore import Qt
from PyQt6.QtGui import QFont, QBrush, QColor
from dataclasses import dataclass

# ─────────────────────────────────────────────────────────────────────────────────────
# DATA MODELS
# ─────────────────────────────────────────────────────────────────────────────────────

@dataclass
class StatMetric:
    """KPI metric card"""
    label: str
    value: str
    icon: str
    trend: str  # UP, DOWN, STABLE
    color: str

@dataclass
class TeamMemberPerformance:
    """Team member sales performance"""
    id: str
    name: str
    role: str
    leadsAssigned: int
    dealsWon: int
    revenue: str
    targetProgress: int  # 0-100%
    status: str  # ACTIVE, INACTIVE

FALLBACK_TEAM = [
    TeamMemberPerformance("t1", "Rajesh Kumar", "Sales Executive", 24, 8, "₹3,20,000", 85, "ACTIVE"),
    TeamMemberPerformance("t2", "Priya Sharma", "Sales Executive", 18, 6, "₹2,10,000", 70, "ACTIVE"),
    TeamMemberPerformance("t3", "Vikram Mehta", "Senior Sales Rep", 32, 11, "₹4,50,000", 95, "ACTIVE"),
    TeamMemberPerformance("t4", "Sunita Rao", "Sales Executive", 15, 5, "₹1,80,000", 60, "ACTIVE"),
    TeamMemberPerformance("t5", "Amit Patel", "Sales Executive", 22, 7, "₹2,80,000", 75, "ACTIVE"),
]

# ─────────────────────────────────────────────────────────────────────────────────────
# STAT CARD COMPONENT
# ─────────────────────────────────────────────────────────────────────────────────────

class StatCard(QFrame):
    """KPI metric display card"""
    def __init__(self, metric: StatMetric, parent=None):
        super().__init__(parent)
        self.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
                padding: 16px;
            }
        """)

        layout = QVBoxLayout(self)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(8)

        # Header with icon and label
        headerLayout = QHBoxLayout()
        iconLabel = QLabel(metric.icon)
        iconLabel.setFont(QFont("Segoe UI", 20))
        headerLayout.addWidget(iconLabel)

        labelWidget = QLabel(metric.label)
        labelWidget.setFont(QFont("Segoe UI", 10))
        labelWidget.setStyleSheet("color: #94a3b8;")
        headerLayout.addWidget(labelWidget)

        # Trend indicator
        trendEmoji = "📈" if metric.trend == "UP" else "📉" if metric.trend == "DOWN" else "➡️"
        trendLabel = QLabel(trendEmoji)
        trendLabel.setFont(QFont("Segoe UI", 12))
        headerLayout.addStretch()
        headerLayout.addWidget(trendLabel)

        layout.addLayout(headerLayout)

        # Value
        valueLabel = QLabel(metric.value)
        valueLabel.setFont(QFont("Segoe UI", 18, QFont.Weight.Bold))
        valueLabel.setStyleSheet(f"color: {metric.color};")
        layout.addWidget(valueLabel)

        layout.addStretch()

# ─────────────────────────────────────────────────────────────────────────────────────
# MAIN MANAGER DASHBOARD VIEW
# ─────────────────────────────────────────────────────────────────────────────────────

class ManagerDashboardView(QWidget):
    """Manager Dashboard with Team Performance Metrics"""

    def __init__(self, parent=None):
        super().__init__(parent)
        self.setStyleSheet("""
            QWidget { background-color: #090d16; }
            QLabel { color: #f8fafc; }
            QPushButton { color: white; font-weight: bold; border: none; border-radius: 6px; }
        """)

        self.teamMembers = list(FALLBACK_TEAM)
        self._build_ui()

    def _build_ui(self):
        """Build manager dashboard UI"""
        layout = QVBoxLayout(self)
        layout.setContentsMargins(0, 0, 0, 0)

        scrollArea = QScrollArea()
        scrollArea.setWidgetResizable(True)
        scrollArea.setStyleSheet("QScrollArea { border: none; background-color: #090d16; }")

        scrollWidget = QWidget()
        scrollLayout = QVBoxLayout(scrollWidget)
        scrollLayout.setContentsMargins(16, 16, 16, 24)
        scrollLayout.setSpacing(12)

        # Title
        titleLabel = QLabel("📊 Manager Dashboard")
        titleLabel.setFont(QFont("Segoe UI", 14, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        scrollLayout.addWidget(titleLabel)

        # KPI Grid
        kpiLayout = QGridLayout()
        kpiLayout.setContentsMargins(0, 0, 0, 0)
        kpiLayout.setSpacing(12)

        kpiMetrics = [
            StatMetric("Team Sales (This Month)", "₹12,40,000", "💰", "UP", "#34d399"),
            StatMetric("Target Progress", "82%", "🎯", "UP", "#60a5fa"),
            StatMetric("Team Size", "5", "👥", "STABLE", "#818cf8"),
            StatMetric("Avg Deal Value", "₹2,48,000", "📈", "UP", "#fbbf24"),
        ]

        for idx, metric in enumerate(kpiMetrics):
            card = StatCard(metric)
            kpiLayout.addWidget(card, idx // 2, idx % 2, 1, 1)

        scrollLayout.addLayout(kpiLayout)

        # Team Performance Section
        teamTitleLayout = QHBoxLayout()
        teamTitleLabel = QLabel("👥 Team Performance")
        teamTitleLabel.setFont(QFont("Segoe UI", 12, QFont.Weight.Bold))
        teamTitleLabel.setStyleSheet("color: #ffffff;")
        teamTitleLayout.addWidget(teamTitleLabel)

        exportBtn = QPushButton("📥 Export Report")
        exportBtn.setStyleSheet("background-color: #4f46e5; padding: 4px 8px; font-size: 9px;")
        exportBtn.clicked.connect(lambda: None)
        teamTitleLayout.addStretch()
        teamTitleLayout.addWidget(exportBtn)

        scrollLayout.addLayout(teamTitleLayout)

        # Team Performance Table
        self.teamTable = QTableWidget()
        self.teamTable.setColumnCount(7)
        self.teamTable.setHorizontalHeaderLabels([
            "Rep Name", "Role", "Leads", "Deals Won", "Revenue", "Target %", "Status"
        ])
        self.teamTable.horizontalHeader().setStretchLastSection(False)
        self.teamTable.setSelectionBehavior(QAbstractItemView.SelectionBehavior.SelectRows)
        self.teamTable.setSelectionMode(QAbstractItemView.SelectionMode.SingleSelection)
        self.teamTable.setStyleSheet("""
            QTableWidget { background-color: #030712; gridline-color: #1e293b; }
            QHeaderView::section { background-color: #0b1329; color: #818cf8; padding: 6px;
                                   border: none; border-right: 1px solid #1e293b; }
        """)

        self.teamTable.setColumnWidth(0, 130)
        self.teamTable.setColumnWidth(1, 120)
        self.teamTable.setColumnWidth(2, 80)
        self.teamTable.setColumnWidth(3, 100)
        self.teamTable.setColumnWidth(4, 120)
        self.teamTable.setColumnWidth(5, 100)
        self.teamTable.setColumnWidth(6, 80)

        self._refresh_team_table()

        scrollLayout.addWidget(self.teamTable, 1)

        scrollArea.setWidget(scrollWidget)
        layout.addWidget(scrollArea)

    def _refresh_team_table(self):
        """Refresh team performance table"""
        self.teamTable.setRowCount(len(self.teamMembers))

        for rowIdx, member in enumerate(self.teamMembers):
            self.teamTable.setItem(rowIdx, 0, QTableWidgetItem(member.name))
            self.teamTable.setItem(rowIdx, 1, QTableWidgetItem(member.role))
            self.teamTable.setItem(rowIdx, 2, QTableWidgetItem(str(member.leadsAssigned)))
            self.teamTable.setItem(rowIdx, 3, QTableWidgetItem(str(member.dealsWon)))
            self.teamTable.setItem(rowIdx, 4, QTableWidgetItem(member.revenue))

            # Target progress bar (visual representation)
            targetItem = QTableWidgetItem(f"{member.targetProgress}%")
            targetColor = "#34d399" if member.targetProgress >= 80 else "#fbbf24" if member.targetProgress >= 60 else "#ef4444"
            targetItem.setForeground(QBrush(QColor(targetColor)))
            self.teamTable.setItem(rowIdx, 5, targetItem)

            statusItem = QTableWidgetItem(member.status)
            statusColor = "#34d399" if member.status == "ACTIVE" else "#ef4444"
            statusItem.setForeground(QBrush(QColor(statusColor)))
            self.teamTable.setItem(rowIdx, 6, statusItem)

File: macOs/Sources/DASCRM/Views/Manager/ManagerDashboardView.swift
"""
ManagerDashboardView.swift — DAS CRM macOS
Manager-Specific KPIs, Team Performance, and Subordinate Metrics
Feature parity with Android ManagerDashboardScreen.tsx
"""

import SwiftUI

struct TeamMemberPerformance: Identifiable {
    let id: String
    let name: String
    let role: String
    let leadsAssigned: Int
    let dealsWon: Int
    let revenue: String
    let targetProgress: Int
    let status: String
}

let fallbackTeamMembers = [
    TeamMemberPerformance(id: "t1", name: "Rajesh Kumar", role: "Sales Executive",
                         leadsAssigned: 24, dealsWon: 8, revenue: "₹3,20,000", targetProgress: 85, status: "ACTIVE"),
    TeamMemberPerformance(id: "t2", name: "Priya Sharma", role: "Sales Executive",
                         leadsAssigned: 18, dealsWon: 6, revenue: "₹2,10,000", targetProgress: 70, status: "ACTIVE"),
    TeamMemberPerformance(id: "t3", name: "Vikram Mehta", role: "Senior Sales Rep",
                         leadsAssigned: 32, dealsWon: 11, revenue: "₹4,50,000", targetProgress: 95, status: "ACTIVE"),
    TeamMemberPerformance(id: "t4", name: "Sunita Rao", role: "Sales Executive",
                         leadsAssigned: 15, dealsWon: 5, revenue: "₹1,80,000", targetProgress: 60, status: "ACTIVE"),
    TeamMemberPerformance(id: "t5", name: "Amit Patel", role: "Sales Executive",
                         leadsAssigned: 22, dealsWon: 7, revenue: "₹2,80,000", targetProgress: 75, status: "ACTIVE"),
]

@MainActor
class ManagerDashboardViewModel: ObservableObject {
    @Published var teamMembers: [TeamMemberPerformance] = fallbackTeamMembers

    var totalTeamSales: String {
        let total = teamMembers.reduce(0) { acc, member in
            let value = Int(member.revenue.replacingOccurrences(of: "₹", with: "").replacingOccurrences(of: ",", with: "")) ?? 0
            return acc + value
        }
        return "₹\(String(format: "%,d", total))"
    }

    var targetProgress: Double {
        Double(teamMembers.map { $0.targetProgress }.reduce(0, +)) / Double(teamMembers.count)
    }

    var avgDealValue: String {
        let totalDeals = teamMembers.reduce(0) { $0 + $1.dealsWon }
        let totalRevenue = teamMembers.reduce(0) { acc, member in
            let value = Int(member.revenue.replacingOccurrences(of: "₹", with: "").replacingOccurrences(of: ",", with: "")) ?? 0
            return acc + value
        }
        let avgValue = totalDeals > 0 ? totalRevenue / totalDeals : 0
        return "₹\(String(format: "%,d", avgValue))"
    }
}

struct ManagerDashboardView: View {
    @StateObject private var viewModel = ManagerDashboardViewModel()
    @State private var selectedMember: TeamMemberPerformance?
    @State private var showMemberDetails = false
    @State private var showExportAlert = false

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Header
            VStack(alignment: .leading, spacing: 12) {
                Text("📊 Manager Dashboard")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(.white)
            }
            .padding(16)
            .background(Color(red: 0.06, green: 0.09, blue: 0.16))
            .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)

            // KPI Grid (2x2)
            VStack(alignment: .leading, spacing: 12) {
                HStack(spacing: 12) {
                    MetricCard(
                        icon: "💰",
                        label: "Team Sales (This Month)",
                        value: viewModel.totalTeamSales,
                        trend: "UP",
                        color: Color(red: 0.2, green: 0.83, blue: 0.60)
                    )

                    MetricCard(
                        icon: "🎯",
                        label: "Target Progress",
                        value: String(format: "%.0f%%", viewModel.targetProgress),
                        trend: "UP",
                        color: Color(red: 0.38, green: 0.65, blue: 0.98)
                    )
                }

                HStack(spacing: 12) {
                    MetricCard(
                        icon: "👥",
                        label: "Team Size",
                        value: String(viewModel.teamMembers.count),
                        trend: "STABLE",
                        color: Color(red: 0.50, green: 0.45, blue: 0.98)
                    )

                    MetricCard(
                        icon: "📈",
                        label: "Avg Deal Value",
                        value: viewModel.avgDealValue,
                        trend: "UP",
                        color: Color(red: 0.98, green: 0.75, blue: 0.14)
                    )
                }
            }
            .padding(16)
            .background(Color(red: 0.06, green: 0.08, blue: 0.14))

            // Team Performance Section
            VStack(alignment: .leading, spacing: 12) {
                HStack {
                    Text("👥 Team Performance")
                        .font(.system(size: 12, weight: .bold))
                        .foregroundColor(.white)

                    Spacer()

                    Button(action: { showExportAlert = true }) {
                        Text("📥 Export Report")
                            .font(.system(size: 9, weight: .bold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 8)
                            .padding(.vertical, 4)
                            .background(Color(red: 0.31, green: 0.27, blue: 0.90))
                            .cornerRadius(4)
                    }
                    .buttonStyle(.plain)
                }

                // Team Performance Table
                Table(viewModel.teamMembers) {
                    TableColumn("Rep Name", value: \.name)
                    TableColumn("Role", value: \.role)
                    TableColumn("Leads") { member in
                        Text(String(member.leadsAssigned))
                    }
                    TableColumn("Deals Won") { member in
                        Text(String(member.dealsWon))
                    }
                    TableColumn("Revenue", value: \.revenue)
                    TableColumn("Target %") { member in
                        Text("\(member.targetProgress)%")
                            .foregroundColor(
                                member.targetProgress >= 80 ? Color(red: 0.2, green: 0.83, blue: 0.60) :
                                member.targetProgress >= 60 ? Color(red: 0.98, green: 0.75, blue: 0.14) :
                                Color(red: 0.98, green: 0.30, blue: 0.40)
                            )
                    }
                    TableColumn("Status") { member in
                        Text(member.status)
                            .foregroundColor(
                                member.status == "ACTIVE" ? Color(red: 0.2, green: 0.83, blue: 0.60) :
                                Color(red: 0.98, green: 0.30, blue: 0.40)
                            )
                    }
                }
                .background(Color(red: 0.03, green: 0.04, blue: 0.07))
                .onDoubleClickSelectAll(false)
            }
            .padding(16)
            .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        }
        .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        .navigationTitle("Manager")
        .alert("📥 Export Report", isPresented: $showExportAlert) {
            Button("OK") { }
        } message: {
            Text("Exporting team performance report...")
        }
    }
}

struct MetricCard: View {
    let icon: String
    let label: String
    let value: String
    let trend: String
    let color: Color

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                Text(icon)
                    .font(.system(size: 20))

                VStack(alignment: .leading, spacing: 2) {
                    Text(label)
                        .font(.system(size: 10))
                        .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                }

                Spacer()

                Text(trend == "UP" ? "📈" : trend == "DOWN" ? "📉" : "➡️")
                    .font(.system(size: 12))
            }

            Text(value)
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(color)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(12)
        .background(Color(red: 0.07, green: 0.09, blue: 0.16))
        .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
        .cornerRadius(12)
    }
}

#Preview {
    NavigationView {
        ManagerDashboardView()
    }
}
